app/controllers/PasskeyAuthFilter.scala

```diff
@@ -0,0 +1,101 @@
+package controllers
+
+import aws.{PasskeyChallengeDB, PasskeyDB}
+import com.gu.googleauth.AuthAction.UserIdentityRequest
+import io.circe.Json
+import logic.Passkey
+import models.JanusException
+import play.api.Logging
+import play.api.http.Status.{BAD_REQUEST, INTERNAL_SERVER_ERROR}
+import play.api.mvc.Results.Status
+import play.api.mvc.{ActionFilter, AnyContent, Result}
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+/** Performs passkey authentication and only allows an action to continue if
+  * authentication is successful.
+  *
+  * See
+  * [[https://webauthn4j.github.io/webauthn4j/en/#webauthn-assertion-verification-and-post-processing]].
+  */
+class PasskeyAuthFilter(host: String)(implicit
+    dynamoDb: DynamoDbClient,
+    ec: ExecutionContext
+) extends ActionFilter[UserIdentityRequest]
+    with Logging {
+
+  def executionContext: ExecutionContext = ec
+
+  def filter[A](request: UserIdentityRequest[A]): Future[Option[Result]] =
+    Future.successful(
```

Comment:
I guess this future is to appease the `ActionFilter` contract?

I'd consider using `Future.apply` here instead of `Future.successful`. `successful` is Future's `pure` method, and is designed to lift a pure value into the Future context. The steps here are actually blocking, and since we're reaching into the Play framework we might not want to be blocking threads. Using `apply` would give Play the opportunity to crack on with other things while the Future returns.

Reply:
Yes it was just to replicate the previous behaviour of the `PasskeyController.authenticate` method while keeping the types happy.  But now moved to a genuine `Future` in 75b26a30a0238e52f80cb2b440256774dc54ebf4